core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -355,18 +355,43 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
    * @param entityName The entityName of the entityType
    * @param configs The config of the entityName
    */
-  def changeConfigs(entityType: String, entityName: String, configs: Properties): Unit = {
+  def changeConfigs(entityType: String, entityName: String, configs: Properties, isUserClientId: Boolean = false): Unit = {
 
     entityType match {
       case ConfigType.Topic => changeTopicConfig(entityName, configs)
       case ConfigType.Client => changeClientIdConfig(entityName, configs)
-      case ConfigType.User => changeUserOrUserClientIdConfig(entityName, configs)
+      case ConfigType.User => changeUserOrUserClientIdConfig(entityName, configs, isUserClientId)
       case ConfigType.Broker => changeBrokerConfig(parseBroker(entityName), configs)
       case ConfigType.Ip => changeIpConfig(entityName, configs)
       case _ => throw new IllegalArgumentException(s"$entityType is not a known entityType. Should be one of ${ConfigType.all}")
     }
   }
 
+  private def tryCleanQuotaNodes(entityType: String, entityName: String, isUserClientId: Boolean = false): Boolean = {
```

Comment:
Could we add some javadoc above this method? And also some inline comments for `isUserClientId` case below, to explain what nodes we're trying to delete for each case. Thanks.

Reply:
Done.